internal/provider/resource_vault_secrets_app.go

```diff
@@ -0,0 +1,164 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	clients "github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func NewVaultSecretsAppResource() *vaultsecretsAppResource {
+	return &vaultsecretsAppResource{}
+}
+
+type vaultsecretsAppResource struct {
+	client *clients.Client
+}
+
+func (r *vaultsecretsAppResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vaultsecrets_app"
+}
+
+func (r *vaultsecretsAppResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			// TODO: Add validators
```

Comment:
This is really good to know. I can definitely add this as an attribute. I'm assuming this would be an optional attribute? Is there an eta on adding project as a resource?

Reply:
Yeah, you can see it in some other resources already: https://github.com/hashicorp/terraform-provider-hcp/blob/main/internal/provider/resource_hvn.go#L86

> Is there an eta on adding project as a resource?
I would guess in the next month or so.

Reply:
nice! That's perfect. I'll add it. Thanks so much